DAT-402 - Fix admin user search bug

madelinekosse opened the pull request:
The call to the search logging endpoint 400'ed when the user was logged in as admin, because the session cookies were being included in the request but the csrf token wasn't there. Since this endpoint doesn't need any authentication, this change omits the credentials from the request and resolves the issue we were seeing, where the redirection  to the dataset page was blocked.

madelinekosse committed:
DAT-402 - Fix admin user search bug

The call to the search logging endpoint 400'ed when the user was logged in as
admin, because the session cookies were being included in the request but the
csrf token wasn't there. Since this endpoint doesn't need any authentication,
this change omits the credentials from the request and resolves the issue we
were seeing, where the redirection  to the dataset page was blocked. As
you can't use "withCredentials = false" for a same site xrh request
(technically you can, but it does nothing), I have switched to a fetch
so the creds are omitted by default.

callum-oakley commented:
Hmm actually still fails for me with the same error 😞 

<img width="649" alt="image" src="https://github.com/GreaterLondonAuthority/dfl-ckanext/assets/11499545/96201ca5-b255-44cb-92c7-e016f7ef353b">

If I set `credentials: "omit"` like you had before then it does work for me (the default is to send cookies for requests on the same origin, of which this is one). Is that where you were seeing the weird slowness?

callum-oakley commented:
I see the slow requests too actually. I notice that if you don't send a cookie CKAN will create a new session and send a new cookie back to you... this shouldn't be slow, but maybe it is? Is there a way we can send the cookie anyway but ignore it? Is that what the decorator Kiran mentioned on slack does?

madelinekosse commented:
> I see the slow requests too actually. I notice that if you don't send a cookie CKAN will create a new session and send a new cookie back to you... this shouldn't be slow, but maybe it is? Is there a way we can send the cookie anyway but ignore it? Is that what the decorator Kiran mentioned on slack does?

Yeah that's the idea, guess I'll try that.

madelinekosse commented:
> Hmm actually still fails for me with the same error disappointed
> <img alt="image" width="649" src="https://user-images.githubusercontent.com/11499545/246875387-96201ca5-b255-44cb-92c7-e016f7ef353b.png">
> 
> If I set `credentials: "omit"` like you had before then it does work for me (the default is to send cookies for requests on the same origin, of which this is one). Is that where you were seeing the weird slowness?

Thanks for testing that out, I had a suspicion it wouldn't work as literally everything I'd tried for this worked sometimes but not other times

madelinekosse committed:
Revert "DAT-402 - Fix admin user search bug"

This reverts commit 2347a90be6f14f197848f7ee014c5dc4d2826169. My
solution did not work in all cases. Adding "credentials: omit" to the
fetch does work but slows the request down too much, so I will use a
different method.

madelinekosse committed:
DAT-402 - Make search log endpoint anonymous access

Rather than not sending the credentials with the request, which was slow, this
fixes the bug by setting the log selected search result endpoint to ignore the
credentials using a toolkit decorator. The docs suggested that this isn't the
correct use case for this decorator, but it seems to work so I went with it.

kjoshi reviewed (approved):
Works Ok for me 👍 

I can search for something, click on a dataset and a new entry ends up the logs CSV. Tested when not logged in, and logged as as non-admin and admin
